src/Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch/ElasticSearchExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// -------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Elasticsearch.Net;
+using Microsoft.Diagnostics.EventFlow.Configuration;
+
+namespace Microsoft.Diagnostics.EventFlow.Outputs.ElasticSearch
+{
+    internal static class ElasticSearchExtensions
+    {
+        internal enum ElasticConnectionPoolType
+        {
+            SingleNode,
+            Static,
+            Sniffing,
+            Sticky
+        }
+
+        internal static IConnectionPool GetConnectionPool(this ElasticSearchOutputConfiguration elasticConfig, IHealthReporter healthReporter)
+        {
+            var esServiceUris = elasticConfig.GetEsServiceUriList(healthReporter);
+            var esConnectionPool = elasticConfig.GetEsConnectionPoolType(healthReporter);
+
+            switch (esConnectionPool)
+            {
+                case ElasticConnectionPoolType.SingleNode:
+                    return new SingleNodeConnectionPool(esServiceUris.First());
+                case ElasticConnectionPoolType.Static:
+                    return new StaticConnectionPool(esServiceUris);
+                case ElasticConnectionPoolType.Sniffing:
+                    return new SniffingConnectionPool(esServiceUris);
+                case ElasticConnectionPoolType.Sticky:
+                    return new StickyConnectionPool(esServiceUris);
+                default:
+                    throw new ArgumentOutOfRangeException($"Invalid ElasticConnectionPoolType: {esConnectionPool}");
+            }
+        }
+
+        private static ElasticConnectionPoolType GetEsConnectionPoolType(
+            this ElasticSearchOutputConfiguration elasticConfig, IHealthReporter healthReporter)
+        {
+            var enumParseResult =
+                Enum.TryParse<ElasticConnectionPoolType>(elasticConfig.ConnectionPoolType, out var elasticConnectionPool);
+            if (enumParseResult)
+            {
+                return elasticConnectionPool;
+            }
+
+            //Invalid config string report and throw
+            var errorMessage = $"{nameof(ElasticSearchOutput)}:  required 'ConnectionPoolType' configuration parameter is invalid. Using default.";
+            healthReporter.ReportProblem(errorMessage, EventFlowContextIdentifiers.Configuration);
+            return ElasticConnectionPoolType.SingleNode;
```

Comment:
I respectfully disagree. Questions to consider:
- How many ES clusters used in production are single node?
- Do we really want to have a default that causes EventFlow to stop sending logs when one node of the cluster goes down?

Reply:
I agree, good point. It would be hard to tell as a user that you badly configured your multi node cluster. 

I will be making the Static connection pool the default. Also making the connection pool optional. And reflect this in the documentation.